### ISSUE TITLE
Asm constraint to require an even register

### ISSUE THREAD

**Issue by asb** [riscv-non-isa/riscv-c-api-doc#37]
We don't currently have an asm constraint to require an even register (for an input or output in a register pair) and I don't believe there's a cross-target constraint already defined. We should likely decide on one. This could be used with the proposed zacas extension (amocas.d on rv32 or amocas.q on rv64), for zdinx, and IIRC the P extension.

**Comment by kito-cheng** [comment]
Good catch, `Zdinx` still using `f` in GCC internally, but I think we should have one for `P` and `Zacas`

**Comment by kito-cheng** [comment]
https://github.com/riscv-non-isa/riscv-c-api-doc/pull/39

**Comment by cmuellner** [comment]
Closing as complete.
